mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -267,21 +268,34 @@ def set_new_roi(self) -> None:
         """
         self.presenter.do_add_roi()
 
-    def update_roi_color_in_table(self, roi_name: str, new_color: tuple):
+    def handle_table_click(self, index) -> None:
+        if index.column() == 1:
+            roi_name = self.roi_table_model.index(index.row(), 0).data()
+            self.spec_roi_color(roi_name)
+
+    def spec_roi_color(self, roi_name: str) -> None:
+        spectrum_roi = self.spectrum_widget.roi_dict.get(roi_name)
+        if spectrum_roi:
+            spectrum_roi.sig_colour_change.connect(self.update_roi_color)
+            spectrum_roi.onChangeColor()
```

Comment:
I think here you want to do: `spectrum_roi.change_color_action.trigger()`. You'll need to make `change_color_action` in `SpectrumROI` be `self.change_color_action`.

Then I'd expect `update_roi_color_in_table()` not too need any changes.